src/PIL/Image.py

```diff
@@ -1531,6 +1531,23 @@ def getpalette(self, rawmode="RGB"):
             rawmode = mode
         return list(self.im.getpalette(mode, rawmode))
 
+    def has_transparency_data(self):
```

Comment:
It would be good to gradually start adding type hints:

```suggestion
    def has_transparency_data(self) -> bool:
```